### ISSUE TITLE
[BUG] PackageValueWithDiscount never set

### ISSUE THREAD

**Issue by LexKoomen** [tig-nl/postnl-magento2#266]
**To Reproduce**
1. Enable the table rates with for example 2 NL rates value = 0, price = 5 and value = 50 price = 0 to make shipping free from 50 euro's or more
2. Set the condition name to 'package_value_with_discount'
3. Add for more than 50 euro in cart
4. Go to shipping step in checkout

**Expected result**
Tig POSTNL shipping rate calculated to 0 because of the table rate

**Actual result**
TIG PostNL shipping rate calculated to 5 euro because the PackageValueWithDiscount is never set in the module.

In vendor/tig/postnl-magento2/Model/ResourceModel/Tablerate.php#120 the bindings are set via a class in module-offline-shipping.

In that class the condition name and value are set according to the condition name which is package_value_with_discount according to the postnl config I have set:

```            
$bind[':condition_name'] = $this->request->getConditionName();
$bind[':condition_value'] = round($this->request->getData($this->request->getConditionName()), 4);
```

So the requested value should be retrieved from $this->request->getData($this->request->getConditionName()) which is:
$this->getData('package_value_with_discount'); but that value is never set in your module.

**Workaround**
Add the $request->setPackageValueWithDiscount() yourself in the request. I guess somewhere in either (or all):
- vendor/tig/postnl-magento2/Service/Carrier/QuoteToRateRequest.php#78
- vendor/tig/postnl-magento2/Service/Carrier/Price/Tablerate.php#100
- vendor/tig/postnl-magento2/Service/Carrier/Price/Tablerate.php#161

** Please complete the following information**
- Magento version: [Magento 2.3.7]
- PHP version: [7.4.0]
- TIG PostNL version: [1.9.9]

**Comment by tig-vincentthart** [comment]
Hello @LexKoomen,

Thank you for the clear description and proposed solution for this case. 
In our next release we will add a fix for this case. Until then you could also switch to the matrix rates within the plugin. 

Greetings,

Vincent

**Comment by LexKoomen** [comment]
Maybe something like:
```
$rateRequest->setPackageValue($this->getValue());
$rateRequest->setPackageValueWithDiscount($this->getValueWithDiscount());
```

```
private function getValueWithDiscount()
{
    $price = array_map(function (Item $item) {
        return $item->getRowTotalInclTax() - $item->getTotalDiscountAmount();
    }, $this->quote->getAllItems());

    return array_sum($price);
}
```

**Comment by LexKoomen** [comment]
Thanks @tig-vincentthart

**Comment by tig-jeffreybranderhorst** [comment]
Hi @LexKoomen ,

This should be fixed in 1.10, our newest version of postnl.

Have a great day,
Jeffrey
